### ISSUE TITLE
On save (Ctrl+S) Updating LatexTools Cache takes a long time

### ISSUE THREAD

**Issue by amiroshn15** [SublimeText/LaTeXTools#1056]
When saving a file (Ctrl+S) the text Updating LatexTools Cache [ = ]" appears in the bottom bar  and stays there for about 60-100 seconds. ST3 becomes frozen. I installed the version 3.12.16 to see what happens there. This does not occur. It seems there is an issue with copying cache on save.

Thank you, 
Alexey.

**Comment by r-stein** [comment]
You can disable the caching [in the settings](https://github.com/SublimeText/LaTeXTools/blob/v3.13.2/LaTeXTools.sublime-settings#L606). I think we should make this opt-in, because it slows down Sublime Text on large documents if you save a often.

**Comment by gerbaulet** [comment]
I can confirm that behaviour; "Updating LaTeXTols cache" takes a long time. I would be interested in what has changed in the last update to evoke this?

**Comment by benruijl** [comment]
Same here. I have many large open latex files, and sublime becomes unusable.

**Comment by r-stein** [comment]
I will just cut a release to disable (opt-in) that feature and we can fix the performance issues afterwards. Actually it should increase the performance by moving calculation stuff in the background.

**Comment by benruijl** [comment]
There are perhaps some small things you can do to improve the situation. For example, only update the cache when the saved file is changed, and allow only one cache update at the same time. If I now press Ctrl+S ten times very quickly, I can cripple sublime.

**Comment by r-stein** [comment]
Yes, that is a good idea. I have just released `v3.13.3` to disable it by default and do nothing if it is disabled. Afterwards we can check that we don't start, if the refresh thread is still running and check for changes.

**Comment by ig0774** [comment]
I've implemented the "don't start, if the refresh thread is still running" bit in #1058. I'm not entirely sure how to go about the "check for changes" bit. The problem is that the view is (obviously) never going to be dirty in the `on_post_save` event. It would be possible to check this in the `on_pre_save` event, but we might have to move to using the non-async versions in order to get the timings rights.

**Comment by r-stein** [comment]
@ig0774 we can use the `view.change_count()` for that (just store the change count of the last time we ran it)

**Comment by ig0774** [comment]
Cool!

**Comment by ig0774** [comment]
Ok, #1058 now uses some logic to check for changes. It's implemented in two separate ways since `view.change_count()` doesn't exist on ST2, but since the async methods don't exist either, I'm not concerned about the timing issues with just checking `view.is_dirty()` in `on_pre_save`.
